src/models/title.rs

```diff
@@ -0,0 +1,90 @@
+use std::error::Error;
+use std::fmt;
+
+#[derive(PartialEq, Debug, Clone, Hash, Eq)]
+/// The title of a [Ticket](Ticket)
+/// Wraps a string and checks that it's not empty when set
+pub struct Title {
+    title: String,
+}
+
+#[derive(PartialEq, Debug, Clone)]
+/// Error if a title cannot be created
+pub struct TitleError {
+    details: String,
+}
+
+/// Sets the error message for a title if it cannot be created
+impl TitleError {
+    fn new(msg: &str) -> TitleError {
+        TitleError {
+            details: msg.to_string(),
+        }
+    }
+}
+
+impl fmt::Display for TitleError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{}", self.details)
+    }
+}
+
+impl Error for TitleError {
+    fn description(&self) -> &str {
+        &self.details
+    }
+}
+
+impl Title {
+    /// Creates a Title for a [Ticket](Ticket)
+    /// results in a [TitleError](TitleError) if the string passed in is empty
+    pub fn new(title: String) -> Result<Title, TitleError> {
+        if title.is_empty() {
+            Err(TitleError::new("Title Cannot be empty"))
+        } else {
+            Ok(Title { title })
+        }
+    }
+
+    /// returns a title for [Ticket](Ticket)
+    pub fn get_title(&self) -> &String {
```

Comment:
Do we need this function at all?

Reply:
It's only used in the two tests below, where we could actually compare directly the `Title` instances now.

Reply:
title is private, so would you go about getting the wrapped string without an accessor?

Reply:
You don't get it - but so far we don't need to get it anywhere either in code or tests.

Reply:
True. I changed it.